doctr/models/modules/transformer/pytorch.py

```diff
@@ -97,6 +100,45 @@ def forward(self, query: torch.Tensor, key: torch.Tensor, value: torch.Tensor, m
         return self.output_linear(x)
 
 
+class EncoderBlock(nn.Module):
+    """Transformer Encoder Block"""
+
+    def __init__(
+        self,
+        num_layers: int,
+        num_heads: int,
+        d_model: int,
+        dropout: float,
+        activation_fct: Callable[[Any], Any] = nn.ReLU(),
+    ) -> None:
+        super().__init__()
+
+        self.num_layers = num_layers
+
+        self.layer_norm = nn.LayerNorm(d_model, eps=1e-5)
+        self.dropout = nn.Dropout(dropout)
+
+        self.attention = nn.ModuleList(
+            [MultiHeadAttention(num_heads, d_model, dropout) for _ in range(self.num_layers)]
+        )
+        self.position_feed_forward = nn.ModuleList(
+            [PositionwiseFeedForward(d_model, d_model, dropout, activation_fct) for _ in range(self.num_layers)]
+        )
+
+    def forward(self, x: torch.Tensor, mask: Optional[torch.Tensor] = None) -> torch.Tensor:
+
+        output = x
+
+        for i in range(self.num_layers):
+            normed_output = self.layer_norm(output)
+            output = output + self.dropout(self.attention[i](normed_output, normed_output, normed_output, mask))
+            normed_output = self.layer_norm(output)
```

Comment:
mmmh I'm not 100% sure, but isn't it supposed to be another layer_norm instance?
Otherwise I imagine that the distribution stats will be biased :thinking:

Reply:
![vit](https://user-images.githubusercontent.com/23098070/189629504-94204815-6205-4b89-9caa-152aecf095f5.png)

Encoder block is the same as in normal transformer only Gelu instead of Relu in MLP

Reply:
Damn i got what you mean... I have to update this also in the decoder :/